src/memory/readUnits.ahk

```diff
@@ -0,0 +1,138 @@
+        ;case "otherPlayers": Offset := 
+        default: return 0
+    }
+    Loop, 128
+    {
+        newOffset := Offset + (8 * (A_Index - 1))
+        arrayAddress := someProcess.BaseAddress + newOffset
+        arrayUnit := someProcess.read(arrayAddress, "Int64")
+        while (arrayUnit > 0 ) { ; keep following the next pointer
+            txtFileNo := someProcess.read(arrayUnit + 0x04, "UInt")
+          
+            if (UnitType:=arrayWatchlist(txtFileNo)) {
+                if (settings["ShowKnownMissileDebug"]){
+                tooltip, % txtFileNo . "", 300, 0, 7
+                }
+                pPath := someProcess.read(arrayUnit + 0x38, "Int64")
+                mode := someProcess.read(arrayUnit + 0x0c, "UInt")
+                unitx := someProcess.read(pPath + 0x02, "UShort")
+                unity := someProcess.read(pPath + 0x06, "UShort")
+                unit := {"txtFileNo": txtFileNo, "x": unitx, "y": unity, "mode": mode, "UnitType": UnitType}
+                array.push(unit)
+                ;if (UnitType = "unknown"){
+                 if (UnitType = "otherMissiles"){
+                    if (settings["ShowOtherMissileDebug"])   
+                    tooltip, % "ID:" . txtFileNo .  " UnitType: " . UnitType, 0, 0
+                    }
+           }   
+            arrayUnit := someProcess.read(arrayUnit + 0x150, "Int64")  ; get next unit
+        }
+    } 
+    return array
+}
+arrayWatchlist(ID) {
+    if ID in %IgnoreMatchlist%
+            return ""
+    if ID in %PhysicalMajorMatchlist%
+        return "PhysicalMajor"
+    if ID in %PhysicalMinorMatchlist%
+            return "PhysicalMinor"
+    if ID in %FireMajorMatchlist%
+            return "FireMajor"
+    if ID in %FireMinorMatchlist%
+            return "FireMinor"
+    if ID in %IceMajorMatchlist%
+            return "IceMajor"
+    if ID in %IceMinorMatchlist%
+            return "IceMinor"
+    if ID in %LightMajorMatchlist%
+            return "LightMajor"
+    if ID in %LightMinorMatchlist%
+            return "LightMinor"
+    if ID in %PoisonMajorMatchlist%
+            return "PoisonMajor"
+    if ID in %PoisonMinorMatchlist%
+            return "PoisonMinor"
+    if ID in %MagicMajorMatchlist%
+            return "MagicMajor"
+    if ID in %MagicMinorMatchlist%
+            return "MagicMinor"
+    if ID in %otherMissilesMatchlist%
+            return "otherMissiles"
+    return ""
+}
+
+buildMatchList(Array){ ;create csv matchlist of Missile ID's for each Catagory
+    for index, param in Array
+        {
+        for j, k in param
+            {
+            if  k is digit 
+                {
+                if (Matchlist = ""  ){
+                        Matchlist:= k
+                    } else {
+                        Matchlist:=Matchlist . "," . k
+                        }
+                } 
+            }
+        }
+    return Matchlist
+}
+buildUnitList(catagory){
+    IniRead, %catagory%, missiles.ini, %catagory%
```

Comment:
This is going to execute for each frame update. Do we want this to happen at startup instead?

Reply:
I'm glad you saw this. This is why Im going to consolidate the script because... AHK. This actually wont update every frame, It will only run once. at the part it was included. AHK literally includes the code AT that location. 

AHK first, reads the script and the first return found not in a function or statement or hotkey ends the autoexecute portion of the script. so everything that we are including is technically ONLY getting executed once. The functions are "loaded into memory" and if the file contains code it will execute if it is within the auto execute.

Since the Include is before your While loop, it is only run once and never touched again... which makes me want to put in a hotkey to turn different LOOT filters on dynamically  💡